tests/test_distro.py

```diff
@@ -432,6 +463,22 @@ def test_cloudlinux7_os_release(self):
         self._test_outcome(desired_outcome)
 
 
+class TestWithRootDir(TestOSRelease):
+    """Test that a LinuxDistribution can be created using an arbitrary root_dir
+    on all OSes.
+    """
+
+    def setup_method(self, test_method):
+        dist = test_method.__name__.split('_')[1]
+        root_dir = os.path.join(DISTROS_DIR, dist)
+        self.distro = distro.LinuxDistribution(
+            os_release_file='',
+            distro_release_file='non',
```

Comment:
Why does it say `'non'` here?

Reply:
That's something I carried over from the test class I subclass here https://github.com/nir0s/distro/blob/ab9d3c1fa7fcc71056700841b6a273b18af20b6d/tests/test_distro.py#L151

I scratched my head as this does not make sense. I think this is meant to be a `non`-existing release file, but I did not dig why you would want that. For the sake of simplicity I kept that as otherwise the tests are not behaving the same way as if you were passing no `distro_release_file` at all. ... which is really weird as this should not matter. @nir0s any clue?

Reply:
I dug a bit in the code as well just now: There is special handling for `distro_release_file` being "true" — `if self.distro_release_file:` — and the method `_parse_distro_release_file` swallows all file access errors in silence. So `non` seems more intentional to me now but I'd use a filename like `no_such_file` or so to better communicate that intention.